overlord/hookstate/ctlcmd/umount.go

```diff
@@ -0,0 +1,80 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package ctlcmd
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/systemd"
+)
+
+var (
+	shortUmountHelp = i18n.G("Undo a temporary or permanent mount")
+	longUmountHelp  = i18n.G(`
+The umount command unmounts the given mount point.`)
+)
+
+func init() {
+	addCommand("umount", shortUmountHelp, longUmountHelp, func() command { return &umountCommand{} })
+}
+
+type umountCommand struct {
+	baseCommand
+	Positional struct {
+		Where string `positional-arg-name:"<where>" required:"yes" description:"path to the mount point"`
+	} `positional-args:"yes" required:"yes"`
+}
+
+func (m *umountCommand) Execute([]string) error {
+	context := m.context()
+	if context == nil {
+		return fmt.Errorf("Cannot mount without a context")
+	}
+
+	if m.Positional.Where == "" {
+		return fmt.Errorf("Mount point cannot be empty")
+	}
+
+	snapName := context.InstanceName()
+
```

Comment:
it's unclear to me what should happen in the following scenario:
1. mount-control is connected
2. snapctl mount <something> happens successfully
3. mount-control is disconnected
4. what does snapctl umount do now?

Reply:
Good question! :-)

I actually wonder if we should perform an umount at point 3.

Also, what should happen if a snap is updated to a new version which is missing some of the the mount-control plugs that are currently mounted? As a user, I would probably expect that these mounts should be umounted at this point.

Reply:
Yeah I'm not sure, what I think should probably happen is to have the disconnection of the interface unmount the mount unit, but let's wait for @pedronis' comment when he's back